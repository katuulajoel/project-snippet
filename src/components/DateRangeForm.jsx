import PropTypes from "prop-types";
import React, { useState } from "react";
import { FormGroup } from "reactstrap";
import moment from "moment";
import styled from "styled-components";

import FieldError from "./FieldError";
import { StyledDateTimePicker, StyledForm } from "./utils/styles";

const DateRangeForm = (props) => {
  const {
    id,
    maxdate: maxdateProp,
    message,
    proceed,
    defaultStart,
    defaultEnd,
  } = props;
  const [start, setstart] = useState(defaultStart || null);
  const [end, setend] = useState(defaultEnd || null);
  const [startError, setstartError] = useState(null);
  const [endError, setendError] = useState(null);
  const maxdate = maxdateProp || new Date(2099, 11, 31);

  const onSave = (e) => {
    e.preventDefault();

    let error = false;
    if (start == null || start == "Invalid date") {
      setstartError("Start date is required");
      setstart(null);
      error = true;
    } else {
      setstartError(null);
    }
    if (end == null || end == "Invalid date") {
      setendError("End date is required");
      setend(null);
      error = true;
    } else {
      setendError(null);
    }

    if (proceed && !error) {
      if (new Date(start) > new Date(end)) {
        setstartError("Start date is greater than end date");
      } else {
        proceed({ start, end });
      }
    }
  };

  return (
    <DateRangeStyledForm id={id} onSubmit={onSave}>
      <FormGroup>
        <label>
          {message}
          <span className="label-style">*</span>
        </label>
      </FormGroup>
      <div>
        <FormGroup>
          {startError && <FieldError message={startError} />}
          <StyledDateTimePicker
            id="start-date"
            className="tg-date-field"
            placeholder="From"
            format={"DD MMM YYYY"}
            calendar={true}
            time={false}
            max={maxdate}
            value={start ? new Date(start) : null}
            onChange={(date) => {
              setstart(moment(date).format());
            }}
            required
          />
        </FormGroup>
        <FormGroup>
          {endError && <FieldError message={endError} />}
          <StyledDateTimePicker
            className="tg-date-field"
            placeholder="To"
            format={"DD MMM YYYY"}
            calendar={true}
            time={false}
            max={maxdate}
            value={end ? new Date(end) : null}
            onChange={(date) => {
              setend(moment(date).format());
            }}
            required
          />
        </FormGroup>
      </div>
    </DateRangeStyledForm>
  );
};

const DateRangeStyledForm = styled(StyledForm)`
  .label-style {
    color: #da3451;
    padding-left: 2px;
  }
  > div {
    display: flex;
    justify-content: space-between;

    > span {
      padding: 10px 16px;
    }
  }
`;

DateRangeForm.propTypes = {
  proceed: PropTypes.func,
  cancel: PropTypes.func,
  dismiss: PropTypes.func,
  id: PropTypes.string,
  message: PropTypes.string,
  maxdate: PropTypes.any,
  defaultStart: PropTypes.string,
  defaultEnd: PropTypes.string,
};

DateRangeForm.defaultProps = {
  message: "Select date range for export",
};

export default DateRangeForm;
