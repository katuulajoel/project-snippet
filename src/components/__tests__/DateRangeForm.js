import React from "react";
import renderer from "react-test-renderer";
import DateRangeForm from "../DateRangeForm";
import mount from "enzyme/build/mount";

describe("Auth layout test", () => {
  it("Snapshot test for NavBar component", () => {
    const tree = renderer.create(<DateRangeForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should submit result", () => {
    const wrapper = mount(<DateRangeForm proceed={() => {}} />);

    console.log(wrapper.debug());

    const startDateInput = wrapper.find("#start-date DatePickerInput input");
    console.log(startDateInput.debug());

    let date = new Date();
    startDateInput.simulate("change", {
      target: { value: date.toISOString() },
    });
  });
});
