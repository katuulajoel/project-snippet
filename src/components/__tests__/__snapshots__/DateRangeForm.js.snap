// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout test Snapshot test for NavBar component 1`] = `
<form
  className="sc-dlnjwi sc-hKFxyN dmLwQU kRXQAI"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label>
      Select date range for export
      <span
        className="label-style"
      >
        *
      </span>
    </label>
  </div>
  <div>
    <div
      className="form-group"
    >
      <div
        calendar={true}
        className="sc-gtsrHT giRYEg tg-date-field rw-date-picker rw-widget"
        data-intent="mouse"
        format="DD MMM YYYY"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        required={true}
        tabIndex={-1}
        time={false}
      >
        <div
          className="rw-widget-picker rw-widget-container"
        >
          <input
            aria-expanded={false}
            aria-haspopup={true}
            aria-owns="start-date_date"
            autoComplete="off"
            className="rw-widget-input rw-input"
            id="start-date_input"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="From"
            role="combobox"
            tabIndex={0}
            type="text"
            value=""
          />
          <button
            aria-label="Select date"
            className="rw-input-addon rw-picker-btn rw-btn"
            onClick={[Function]}
            tabIndex={-1}
            title="Select date"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="rw-i-calendar"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        calendar={true}
        className="sc-gtsrHT giRYEg tg-date-field rw-date-picker rw-widget"
        data-intent="mouse"
        format="DD MMM YYYY"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        required={true}
        tabIndex={-1}
        time={false}
      >
        <div
          className="rw-widget-picker rw-widget-container"
        >
          <input
            aria-expanded={false}
            aria-haspopup={true}
            aria-owns="rw_4_date"
            autoComplete="off"
            className="rw-widget-input rw-input"
            id="rw_3_input"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="To"
            role="combobox"
            tabIndex={0}
            type="text"
            value=""
          />
          <button
            aria-label="Select date"
            className="rw-input-addon rw-picker-btn rw-btn"
            onClick={[Function]}
            tabIndex={-1}
            title="Select date"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="rw-i-calendar"
              fill="currentcolor"
              height="1em"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
